### ISSUE TITLE
Using COLLAB Dataset

### ISSUE THREAD

**Issue by pixar0407** [rusty1s/pytorch_geometric#1544]
## ❓ Questions & Help

hello.

I want to use collab dataset. 
I could find the way of pubmed, citeseer, cora using Plaintoid.
And I could find the way of reddit using "from torch_geometric.datasets import Reddit" 

But I cannot find the way of using "collab dataset" or any example code of using "collab dataset".

In this web page(https://ls11-www.cs.tu-dortmund.de/staff/morris/graphkerneldatasets), this says 
"The datasets can also be accessed using PyTorch Geometric" which makes me think that there is a certain way of calling "collab dataset".

Is there any way OR i ought to make my own using raw collab data?
thank you.

**Comment by rusty1s** [comment]
`COLLAB` can be accessed via `torch_geometric.datasets.TUDataset` :)

**Comment by pixar0407** [comment]
Oh! thank you for reply. then, two following small detail questions.

Q1. the possible 'name' values could be found in this web page(https://chrsmrrs.github.io/datasets/docs/datasets/), right?

upper-case letter and lower-case letter(english alphabet) should be decided by the list of names in this web page, right?

Q2. Then, simply this would works, right? 
from torch_geometric.datasets import TUDataset
dataset = TUDataset(path, name='COLLAB')

Thank you :)

**Comment by rusty1s** [comment]
Exactly!

**Comment by pixar0407** [comment]
hello, again.

I could get COLLAB 5000 dataset, as you instructed.
But the datasets look like this (ex, Data(edge_index=[2,2352], y=[1])), which means this dataset is not for graph neural network. It seems normal graph data.

What i want was **COLLAB dataset for Graph Neural Network**.
Dataset's format is like this (#vertex 12,087 #feature length 492, #edge 1,446,010 from this paper https://arxiv.org/pdf/2001.02514.pdf).
After i googled, I found this site(https://dl.acm.org/doi/pdf/10.1145/2783258.2783417) would be related to it.

Can you tell me how to get COLLAB for graph neural network.
Thank you.

**Comment by pixar0407** [comment]
Oh, i will try this(https://github.com/rusty1s/pytorch_geometric/issues/1557).
Thanks.

**Comment by rusty1s** [comment]
It seems like they are using exactly this dataset, but batch it together into a single graph. Node features need to be added manually.

**Comment by pixar0407** [comment]
okay. dataset[0] is like "Data(edge_index=[2,2352], y=[1])". So, it seems like graph classification problem. Because one data has bunch of edges but no vertex label, only has one label. I wonder how i can decide the size of vertices. But i will try and find the solution and share that in this issue later.

Thanks!
